cmd/fpga_tool/fpga_tool.go

```diff
@@ -247,21 +247,34 @@ func printFpgaPort(f fpga.Port, quiet bool) (err error) {
 	fmt.Printf("Port Id                          : %d\n", num)
 	fmt.Printf("Interface UUID                   : %s\n", f.GetInterfaceUUID())
 	fmt.Printf("Accelerator UUID                 : %s\n", f.GetAcceleratorTypeUUID())
-	if !quiet {
-		if apiVer, err := f.GetAPIVersion(); err == nil {
-			fmt.Printf("Kernet API Version               : %d\n", apiVer)
-			pi, err := f.PortGetInfo()
-			if err == nil {
-				fmt.Printf("Port Regions                     : %d\n", pi.Regions)
-				for idx := 0; uint32(idx) < pi.Regions; idx++ {
-					if ri, err := f.PortGetRegionInfo(uint32(idx)); err == nil {
-						fmt.Printf("Port Region (Index/Size/Offset)  : %d / %d / %d\n", ri.Index, ri.Size, ri.Offset)
-					}
-				}
-			}
+
+	if quiet {
+		return err
+	}
+
+	apiVer, err := f.GetAPIVersion()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Kernet API Version               : %d\n", apiVer)
+
+	pi, err := f.PortGetInfo()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Port Regions                     : %d\n", pi.Regions)
+	for idx := uint32(0); idx < pi.Regions; idx++ {
+		if ri, err2 := f.PortGetRegionInfo(idx); err2 == nil {
+			fmt.Printf("Port Region (Index/Size/Offset)  : %d / %d / %d\n", ri.Index, ri.Size, ri.Offset)
+		} else {
+			// Keep iterating, but don't loose at least the last error.
+			err = err2
```

Comment:
I think we can return error immediately. If earlier ioctl call to driver fails, there is very minor chance that subsequent will succeed

Reply:
Yeap, updated to return immediately.